src/main/java/edu/wpi/first/pathweaver/PropertyManager.java

```diff
@@ -0,0 +1,215 @@
+    valueCol.setCellFactory(param -> new EditingCell());
+
+    valueCol.setOnEditCommit(
+        t -> {
+          int row = t.getTablePosition().getRow();
+          NamedProperty property = t.getTableView().getItems().get(row);
+          Object oldVal = t.getOldValue();
+          property.coerceValue(t.getNewValue());
+          if (!commitCallback.call(current)) {
+            property.coerceValue(oldVal);
+          }
+        }
+    );
+    this.propertyView.getColumns().add(nameCol);
+    this.propertyView.getColumns().add(valueCol);
+  }
+
+  /**
+   * Sets the TableView to show the editable properties of the object, or deselects editing if null is passed.
+   * @param object The object to show and edit the properties of
+   */
+  public void manageProperties(PropertyEditable object) {
+    current = object;
+    if (object == null) {
+      this.propertyView.setItems(null);
+    } else {
+      this.propertyView.setItems(object.getProperties());
+    }
+    this.propertyView.refresh();
+  }
+
+  /**
+   * Sets the callback to be called when the PropertyManager tries to commit a property. Should validate and cleanup.
+   * @param commitCallback The callback, where the first argument is the object that was just edited
+   *                           and the return value is whether the edit should go ahead.
+   */
+  public void setCommitCallback(Callback<PropertyEditable, Boolean> commitCallback) {
+    this.commitCallback = commitCallback;
+  }
+
+  public static class NamedProperty {
+    private final Property value;
+    private final SimpleStringProperty name = new SimpleStringProperty();
+
+    public NamedProperty(String name, Property value) {
+      this.name.set(name);
+      this.value = value;
+    }
+
+    public void setValue(Object val) {
+      value.setValue(val);
+    }
+
+    /**
+     * Attempts to coerce val, first by setting the property, then by coercing to Double, then failing.
+     * @param val The value to try to coerce.
+     */
+    public void coerceValue(Object val) {
+      try {
+        value.setValue(val);
+        return;
+      } catch (ClassCastException ignored) { }
+
+      String sval = (String) val;
+      try {
+        value.setValue(Double.parseDouble(sval));
+        return;
+      } catch (ClassCastException ignored) { }
+
+      throw new IllegalArgumentException("Failed to coerce property");
+    }
+
+    public StringProperty nameProperty() {
+      return name;
+    }
+
+    public Property valueProperty() {
+      return value;
+    }
+  }
+
+  public interface PropertyEditable {
+    ObservableList<NamedProperty> getProperties();
+  }
+
+  private class EditingCell extends TableCell<NamedProperty, Object> {
+    private TextField textField;
+    private CheckBox checkBox;
+
+    public EditingCell() {
+      super();
+    }
+
+    @Override public void startEdit() {
+      if (!isEmpty()) {
+        super.startEdit();
+
+        if (getItem() instanceof Boolean) {
```

Comment:
Possibly, but why?

Reply:
Eh, it's usually bad practice, i.e. when using instanceof, code is very tightly aligned with types rather than allowing for new data types later.

Reply:
For example, if I type "abc" into any of the fields, right now it will throw an `IllegalArgumentException("Failed to coerce property")` but in the future if we have a string field, it will properly convert, which would not be the expected behavior for a double field.

Reply:
Additionally, we should probably not throw that exception.

Reply:
The problem is that this section specifically where it creates the CheckBox or the TextField has to be tightly aligned with types, because the Boolean for a CheckBox is a very special case. I don't see any other cases being introduced, because it's much easier to represent/coerce other types as a String, and thus the TextField is sufficient.
The coercion code is separate (L 115), and while that's not ideal, there's no general-purpose way to coerce a String into other types. Again I don't see many more cases being introduced, as this code will work with Doubles, Strings, and Booleans.
The IllegalArgumentException has been caught.